src/utility/util.jl

```diff
@@ -45,53 +45,49 @@ function _elementwise_mult(a₁::AbstractTensorMap, a₂::AbstractTensorMap)
     return dst
 end
 
-_safe_pow(a, pow, tol) = (pow < 0 && abs(a) < tol) ? zero(a) : a^pow
+_safe_pow(a::Real, pow::Real, tol::Real) = (pow < 0 && abs(a) < tol) ? zero(a) : a^pow
 
 """
-    sdiag_pow(S::AbstractTensorMap, pow::Real; tol::Real=eps(scalartype(S))^(3 / 4))
+    sdiag_pow(S::DiagonalTensorMap, pow::Real; tol::Real=eps(scalartype(S))^(3 / 4))
 
-Compute `S^pow` for diagonal matrices `S`.
+Compute `S^pow` for DiagonalTensorMap `S`.
```

Comment:
I think we could simply add a specialization for `DiagonalTensorMap` instead of removing the old implementation.